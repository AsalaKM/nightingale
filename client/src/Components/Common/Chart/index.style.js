import styled from "styled-components";

export const PieChart = styled.div`
  display: block;
  width: 60%;
  margin: 0 auto;
`;

export const Header = styled.h4`
  font-size: 20px;
  color: #333333;
  margin-top: 2rem;
`;

export const Charts = styled.div`
  width: 100%;
  margin: 0 auto;
  margin-top: 2rem;
  display: flex;
  justify-content: space-between;
`;

export const Title = styled.div`
    animation: up 0.5s ease-in 0.5s;
    animation-fill-mode: both;
    font-size: 13px;
    color:#555555;


  @keyframes up {
    0% {
      visibility: visible;
      opacity: 0;
      transform: translateY(1rem);
    }
    70% {
      transform: translateY(-0.1rem);
    }
    100% {
      opacity: 1;
      transform: translate(0);
    }
`;
export const Decription = styled.span`
  display: inline-block;
  width: 30px;
  height: 15px;
  margin-right: 30px;
`;
export const Svg = styled.svg`
  margin-top: 4rem;
  animation: down 0.7s ease-in;

  @keyframes down {
    0% {
      opacity: 0;
      transform: translateY(-1rem);
    }
    70% {
      transform: translateY(0.1rem);
    }
    100% {
      opacity: 1;
      transform: translate(0);
    }
  }
`;

export const Img = styled.img``;
