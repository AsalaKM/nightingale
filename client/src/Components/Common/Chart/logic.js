import * as d3 from 'd3';

export const layout = [
  '#6b486b', '#a05d56', '#d0743c', '#ff8c00','#98abc5', '#8a89a6', '#7b6888',
];

export const Chart = (sections, tag, width) => {
  const svgWidth = width;

  const svgHeight = width;

  const radius = Math.min(svgWidth, svgHeight) / 2;

  const svg = d3
    .select(tag)
    .attr('width', svgWidth)
    .attr('height', svgHeight);

  const group = svg
    .append('g')
    .attr('transform', `translate(${radius},${radius})`);

  const pie = d3.pie().value(d => d.percentage);

  const path = d3
    .arc()
    .outerRadius(radius)
    .innerRadius(radius - 50);

  const arc = group
    .selectAll('arc')
    .data(pie(sections))
    .enter()
    .append('g');

  arc
    .append('path')
    .attr('d', path)
    .attr('fill', (d, i) => layout[i]);

  const label = d3
    .arc()
    .outerRadius(radius)
    .innerRadius(radius - 50);

  arc
    .append('text')
    .attr('transform', d => `translate(${label.centroid(d)})`)
    .attr('text-anchor', 'middle')
    .text(d => `${d.data.percentage} %`)
    .style('font-size', '15px')
    .style('font-weight', 'bold')
    .style('fill', 'white')
    .style('cursor', 'pointer')
    .append('svg:title')
    .text(d => d.data.decription);
};
