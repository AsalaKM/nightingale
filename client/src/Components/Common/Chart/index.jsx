import React, { Component } from "react";
import uuid from "uuid";
import { Chart, layout } from "./logic";
import {
  PieChart,
  Header,
  Charts,
  Title,
  Decription,
  Svg,
  Img
} from "./index.style";

export default class Pie extends Component {
  constructor(props) {
    super(props);
    this.chartTag = React.createRef();
  }

  componentDidMount = () => {
    const { sections, width } = this.props;
    Chart(sections, this.chartTag.current, width);
  };

  render() {
    const { sections, id, width, title } = this.props;
    return (
      <PieChart>
        <Header>{title}</Header>
        <Svg
          className={`bar-chart-${id} bar-chart`}
          ref={this.chartTag}
          style={{ width }}
        />
        <Charts>
          {sections.map((section, index) => (
            <Title key={uuid()}>
              <Decription
                style={{
                  background: layout[index],
                  width: "15px",
                  height: "15px",
                  marginRight: "12px"
                }}
              />
              <Img src={section.decription} alt="emoji" />
            </Title>
          ))}
        </Charts>
      </PieChart>
    );
  }
}
