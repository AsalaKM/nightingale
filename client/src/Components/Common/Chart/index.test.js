import React from "react";
import renderer from "react-test-renderer";
import Chart from "./index.jsx";

describe("Chart", () => {
  it("renders correctly", () => {
    const tree = renderer
      .create(
        <Chart
          sections={[
            { decription: "title 1", percentage: 20 },
            { decription: "title 1", percentage: 30 },
            { decription: "title 1", percentage: 25 },
            { decription: "title 1", percentage: 25 }
          ]}
          width={200}
          title="Mood Outcomes"
        />
      )
      .toJSON();
    expect(tree).toMatchSnapshot();
  });
});
