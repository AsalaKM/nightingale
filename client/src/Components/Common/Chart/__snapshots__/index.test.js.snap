// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Chart renders correctly 1`] = `
<div
  className="sc-bdVaJa ehdsTR"
>
  <h4
    className="sc-bwzfXH gqJssl"
  >
    Mood Outcomes
  </h4>
  <svg
    className="bar-chart-undefined bar-chart sc-EHOje iqMUOf"
    style={
      Object {
        "width": 200,
      }
    }
  />
  <div
    className="sc-htpNat jbLzsI"
  >
    <div
      className="sc-bxivhb foBzPx"
    >
      <p>
        title 1
      </p>
      <span
        className="sc-ifAKCX csFauE"
        style={
          Object {
            "color": "#6b486b",
            "fontSize": "13px",
            "fontWeight": "600",
            "textTransform": "capitalize",
          }
        }
      />
      <span
        className="sc-bZQynM jsjzYN"
      >
        (
        )
      </span>
    </div>
    <div
      className="sc-bxivhb foBzPx"
    >
      <p>
        title 1
      </p>
      <span
        className="sc-ifAKCX csFauE"
        style={
          Object {
            "color": "#a05d56",
            "fontSize": "13px",
            "fontWeight": "600",
            "textTransform": "capitalize",
          }
        }
      />
      <span
        className="sc-bZQynM jsjzYN"
      >
        (
        )
      </span>
    </div>
    <div
      className="sc-bxivhb foBzPx"
    >
      <p>
        title 1
      </p>
      <span
        className="sc-ifAKCX csFauE"
        style={
          Object {
            "color": "#d0743c",
            "fontSize": "13px",
            "fontWeight": "600",
            "textTransform": "capitalize",
          }
        }
      />
      <span
        className="sc-bZQynM jsjzYN"
      >
        (
        )
      </span>
    </div>
    <div
      className="sc-bxivhb foBzPx"
    >
      <p>
        title 1
      </p>
      <span
        className="sc-ifAKCX csFauE"
        style={
          Object {
            "color": "#ff8c00",
            "fontSize": "13px",
            "fontWeight": "600",
            "textTransform": "capitalize",
          }
        }
      />
      <span
        className="sc-bZQynM jsjzYN"
      >
        (
        )
      </span>
    </div>
  </div>
</div>
`;
